proxy/envoy/config_test.sh

```diff
@@ -0,0 +1,22 @@
+#!/bin/bash
+# Check that envoy can parse golden configuration artifacts
+# In the future, we can run functional tests, but all configs require iptables rules to be effective.
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+ENVOY="proxy/envoy/envoy -l trace --service-cluster istio-proxy --service-node 10.0.0.1"
+CONFIG_FILES="envoy-v0 envoy-fault ingress-envoy egress-envoy"
+
+# TODO the test skips over SSL configuration files since referenced cert files are not in the right places
+
+for f in ${CONFIG_FILES}; do
+  echo "##############################"
+  echo "# Testing $f..."
+  echo "##############################"
+  ${ENVOY} -c proxy/envoy/testdata/$f.json.golden &
+  PID=$!
+  sleep 1
```

Comment:
> ... wait till Envoy completes a round of *DS fetches

In that case, what about curl'ing the `localhost:<admin-port>/stats` and grepping for specific SDS/CDS/RDS stats to verify envoy has populated them correctly after polling discovery service? Envoy's discovery interval can be 2 * refresh_delay_ms, e.g. 2 seconds, in which case this test wouldn't sufficiently verify that one round of SDS/CDS/RDS completed successfully.